nixos/modules/services/misc/geoipupdate.nix

```diff
@@ -99,9 +99,22 @@ in
       LockFile = "/run/geoipupdate/.lock";
     };
 
+    systemd.services.geoipupdate-create-db-dir = {
+      serviceConfig.Type = "oneshot";
+      script = ''
+        mkdir -p ${cfg.settings.DatabaseDirectory}
+        chmod 0755 ${cfg.settings.DatabaseDirectory}
+      '';
+    };
+
     systemd.services.geoipupdate = {
       description = "GeoIP Updater";
-      after = [ "network-online.target" "nss-lookup.target" ];
+      requires = [ "geoipupdate-create-db-dir.service" ];
+      after = [
+        "geoipupdate-create-db-dir.service"
+        "network-online.target"
```

Comment:
Does this service handle gracefully when there is not internet connection? I would prefer a network.target here.

Reply:
I haven't tried running it without network, but I would imagine it just fails to run and that's that. I really want the first run to be when the network is available, though, and there's no point in running the service until a connection has been established, since it's guaranteed to fail. This seems to match the intended use case for the target well, reading its section at https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/:

> ...its primary purpose is network client software that cannot operate without network